hack/setup.sh

```diff
@@ -201,6 +245,15 @@ function createBackstageSecret {
   if [ -n "$GITHUB_CLIENT_SECRET" ]; then
     secretKeys[GITHUB_CLIENT_SECRET]=$GITHUB_CLIENT_SECRET
   fi
+  if [ -n "$NOTIFICATIONS_EMAIL_HOSTNAME" ]; then
```

Comment:
shouldn't we check for `SETUP_NOTIFICATIONS_EMAIL` to be `true` as well? The check already exists for capturing the values, and if `SETUP_NOTIFICATIONS_EMAIL`==false && the env env has the keys defined, they will be included in the secret.

Reply:
there is a condition not to invoke this function in line 332.
do you think we should add another safety?

Reply:
Thanks @jordigilh for spotting this. I change the condition.